GVRf/Extensions/MixedReality/src/main/java/org/gearvrf/mixedreality/GVRMixedReality.java

```diff
@@ -0,0 +1,270 @@
+        mSession.registerAugmentedImageListener(listener);
+    }
+
+    @Override
+    public ArrayList<GVRPlane> getAllPlanes() {
+        if (mState == SessionState.ON_PAUSE) {
+            throw new UnsupportedOperationException("Session is not resumed");
+        }
+        return mSession.getAllPlanes();
+    }
+
+
+    @Override
+    public GVRAnchor createAnchor(float[] pose) {
+        if (mState == SessionState.ON_PAUSE) {
+            throw new UnsupportedOperationException("Session is not resumed");
+        }
+        return mSession.createAnchor(pose);
+    }
+
+    @Override
+    public GVRAnchor createAnchor(float[] pose, GVRSceneObject sceneObject) {
+        if (mState == SessionState.ON_PAUSE) {
+            throw new UnsupportedOperationException("Session is not resumed");
+        }
+        return mSession.createAnchor(pose, sceneObject);
+    }
+
+    @Override
+    public void updateAnchorPose(GVRAnchor anchor, float[] pose) {
+        if (mState == SessionState.ON_PAUSE) {
+            throw new UnsupportedOperationException("Session is not resumed");
+        }
+        mSession.updateAnchorPose(anchor, pose);
+    }
+
+    @Override
+    public void removeAnchor(GVRAnchor anchor) {
+        if (mState == SessionState.ON_PAUSE) {
+            throw new UnsupportedOperationException("Session is not resumed");
+        }
+        mSession.removeAnchor(anchor);
+    }
+
+    @Override
+    public void hostAnchor(GVRAnchor anchor, ICloudAnchorListener listener) {
+        mSession.hostAnchor(anchor, listener);
+    }
+
+    @Override
+    public void resolveCloudAnchor(String anchorId, ICloudAnchorListener listener) {
+        mSession.resolveCloudAnchor(anchorId, listener);
+    }
+
+    @Override
+    public void setEnableCloudAnchor(boolean enableCloudAnchor) {
+        mSession.setEnableCloudAnchor(enableCloudAnchor);
+    }
+
+    @Override
+    public GVRHitResult hitTest(GVRSceneObject sceneObj, GVRPicker.GVRPickedObject collision) {
+        if (mState == SessionState.ON_PAUSE) {
+            throw new UnsupportedOperationException("Session is not resumed");
+        }
+        return mSession.hitTest(sceneObj, collision);
+    }
+
+    @Override
+    public GVRLightEstimate getLightEstimate() {
+        if (mState == SessionState.ON_PAUSE) {
+            throw new UnsupportedOperationException("Session is not resumed");
+        }
+        return mSession.getLightEstimate();
+    }
+
+    @Override
+    public void setAugmentedImage(Bitmap image) {
+        mSession.setAugmentedImage(image);
+    }
+
+    @Override
+    public void setAugmentedImages(ArrayList<Bitmap> imagesList) {
+        mSession.setAugmentedImages(imagesList);
+    }
+
+    @Override
+    public ArrayList<GVRAugmentedImage> getAllAugmentedImages() {
+        return mSession.getAllAugmentedImages();
+    }
+
+    private class ActivityEventsHandler implements IActivityEvents {
```

Comment:
GVREventListeners defines ActivityEvents - empty implementation of IActivityEvents. You could derive from that in the future.